dali/python/nvidia/dali/plugin/base_iterator.py

```diff
@@ -226,59 +227,60 @@ def __init__(
                 )
 
         if self._enable_checkpointing:
-            if self._last_batch_policy == LastBatchPolicy.FILL and self._last_batch_padded is False:
-                raise NotImplementedError(
-                    "Currently, checkpointing is not supported for iterators with "
-                    + "last_batch_policy=FILL and last_batch_padded=False"
-                )
-
             # Precompute the initial checkpoints, to prevent any problems
             # related to the `prepare_first_batch` flag.
             self._initial_checkpoints = [p.checkpoint() for p in self._pipes]
```

Comment:
That one needs be adjusted too, not to skip iterator state if there's any, doesn't it?

Would moving it just before `self._restore_state(iterator_data)` and making it `self._initial_checkpoints = [p._get_checkpoint(iterator_data) for p in self._pipes]` work?

Reply:
That's a nice catch and we don't have a test for that. I'll fix this add a test with multiple save+restore

Reply:
Fixed in https://github.com/NVIDIA/DALI/pull/5371/commits/d13c0f6bfabfb5d5da97bd7205b36f7030cfc868

Reply:
Thank you!